docs/hello_nextflow/02_hello_channels.md

```diff
@@ -0,0 +1,868 @@
+N E X T F L O W  ~  version 24.02.0-edge
+Launching `hello-channels.nf` [big_woese] DSL2 - revision: 53f20aeb70
+[62/d81e63] Submitted process > sayHello (1)
+[19/507af3] Submitted process > sayHello (2)
+[8a/3126e6] Submitted process > sayHello (3)
+```
+
+That's much better; at least for a simple workflow.
+For a complex workflow, or a large number of inputs, having the full list output to the terminal might get a bit overwhelming.
+
+That being said, we have another problem. If you look in the `results` directory, there is only one file: `output.txt`!
+
+```console title="Directory contents"
+results
+└── output.txt
+```
+
+What's up with that? Shouldn't we be expecting a separate file per input greeting, so three files in all?
+Did all three greetings go into a single file?
+You can check the contents of `output.txt`; you will find only one of the three.
+
+You may recall that we hardcoded the output file name for the `sayHello` process, so all three calls produced a file called `output.txt`.
+You can check the work subdirectories for each of the three processes; each of them contains a file called `output.txt` as expected.
+
+As long as the output files stay there, isolated from the other processes, that is okay.
+But when the `publishDir` directive copies each of them to the same `results` directory, whichever got copied there first gets overwritten by the next one, and so on.
+
+### 2.4. Ensure the output file names will be unique
+
+We can continue publishing all the outputs to the same results directory, but we need to ensure they will have unique names.
+Specifically, we need to modify the first process to generate a file name dynamically so that the final file names will be unique.
+
+So how do we make the file names unique? A common way to do that is to use some unique piece of metadata as part of the file name.
+Here, for convenience, we'll just use the greeting itself since it's just a short string, and prepend it to the base output filename.
+
+In the process block, make the following code changes:
+
+_Before:_
+
+```groovy title="hello-channels.nf" linenums="11"
+process sayHello {
+
+    publishDir 'results', mode: 'copy'
+
+    input:
+        val greeting
+
+    output:
+        path 'output.txt'
+
+    script:
+    """
+    echo '$greeting' > output.txt
+    """
+}
+```
+
+_After:_
+
+```groovy title="hello-channels.nf" linenums="11"
+process sayHello {
+
+    publishDir 'results', mode: 'copy'
+
+    input:
+        val greeting
+
+    output:
+        path "${greeting}-output.txt"
+
+    script:
+    """
+    echo '$greeting' > '$greeting-output.txt'
+    """
+}
+```
+
+Make sure to replace `output.txt` in both the output definition and in the `script:` command block.
+
+!!! tip
+
+    In the output definition, you MUST use double quotes around the output filename expression (NOT single quotes), otherwise it will fail.
+
+This should produce a unique output file name every time the process is called.
```

Comment:
```suggestion
This should produce a unique output file name every time the process is called, so that it can be distinguished from the outputs from other iterations of the same process in the output directory.
```
Just doubling down, not sure if it's helpful.